metaflow/datatools/s3.py

```diff
@@ -931,12 +931,19 @@ def _one_boto_op(self, op, url, create_tmp_file=True):
             if tmp:
                 os.unlink(tmp.name)
             self._s3_client.reset_client()
-            # add some jitter to make sure retries are not synchronized
-            time.sleep(2 ** i + random.randint(0, 10))
+            # only sleep if retries > 0
+            if S3_RETRY_COUNT > 0:
+                self._jitter_sleep(i)
         raise MetaflowS3Exception(
             "S3 operation failed.\n" "Key requested: %s\n" "Error: %s" % (url, error)
         )
 
+    # add some jitter to make sure retries are not synchronized
+    def _jitter_sleep(self, trynum, multiplier=2):
+        interval = multiplier**trynum + random.randint(0, 10)
+        os.environ["S3_RETRY_INTERVAL"] = str(interval)
```

Comment:
What is the purpose of this here? Is it so you can monitor it?

Reply:
DRYing up the code to avoid duplicating sleep w/jitter logic in multiple places.

Reply:
What changes would you like to see w/this PR? Remove the function? Does `$S3_RETRY_INTERVAL` need to be set?

Reply:
Apologies, I understand the use of the function `_jitter_sleep`; I was just curious about the setting of the environment variable that is not used anywhere else as far as I can tell. In other words, if you don't need it for something, I would just strip it out.